core/src/main/java/io/undertow/conduits/ChunkedStreamSinkConduit.java

```diff
@@ -200,6 +200,9 @@ int doWrite(final ByteBuffer src) throws IOException {
 
     @Override
     public void truncateWrites() throws IOException {
+        if (anyAreSet(state, FLAG_FINISHED)) {
+            return;
+        }
```

Comment:
I think we have another subtle bug in this method. The `invokeFinishListener()` is called prior to the `finally`, so the connection may already be used by another thread by the time we call `truncateWrites`. I'll push an update shortly.

Reply:
On second thought, this shouldn't matter in practice because terminateWrites will close the underlying persistent connection, so we don't need to worry about reuse.